xrpl/models/transactions/batch.py

```diff
@@ -0,0 +1,158 @@
+            else:
+                tx_dict = tx.to_dict()
+                if "flags" not in tx_dict:
+                    tx_dict["flags"] = TransactionFlag.TF_INNER_BATCH_TXN
+                elif isinstance(tx_dict["flags"], int):
+                    tx_dict["flags"] += TransactionFlag.TF_INNER_BATCH_TXN
+                elif isinstance(tx_dict["flags"], dict):
+                    tx_dict["flags"]["TF_INNER_BATCH_TXN"] = True
+                elif isinstance(self.flags, list):
+                    tx_dict["flags"].append(TransactionFlag.TF_INNER_BATCH_TXN)
+                elif tx_dict["flags"] is None:
+                    tx_dict["flags"] = TransactionFlag.TF_INNER_BATCH_TXN
+                else:
+                    raise XRPLModelException(
+                        "Invalid `flags` type for raw transaction in Batch."
+                    )
+                new_raw_transactions.append(Transaction.from_dict(tx_dict))
+        # This is done before dataclass locks in the frozen fields
+        # This way of editing is a bit hacky, but it's the recommended method for
+        # updating frozen fields in dataclasses
+        # https://docs.python.org/3/library/dataclasses.html#frozen-instances
+        object.__setattr__(self, "raw_transactions", new_raw_transactions)
+        super().__post_init__()
+
+    def _get_errors(self: Self) -> Dict[str, str]:
+        errors = super()._get_errors()
+        for i in range(len(self.raw_transactions)):
+            tx = self.raw_transactions[i]
+            if not tx.has_flag(TransactionFlag.TF_INNER_BATCH_TXN):
+                errors[f"raw_transactions[{i}]"] = (
+                    "RawTransaction must have tfInnerBatchTxn flag set."
+                )
+        if len(self.raw_transactions) < 2:
+            errors["raw_transactions"] = "Batch must contain at least 2 transactions."
+        return errors
+
+    @classmethod
+    def from_dict(cls: Type[Self], value: Dict[str, Any]) -> Self:
+        """
+        Construct a new Batch from a dictionary of parameters.
+
+        Args:
+            value: The value to construct the Batch from.
+
+        Returns:
+            A new Batch object, constructed using the given parameters.
+
+        Raises:
+            XRPLModelException: If the dictionary provided is invalid.
+        """
+        new_value = {**value}
+        new_value["raw_transactions"] = [
+            (
+                tx["raw_transaction"]
+                if isinstance(tx, dict) and "raw_transaction" in tx
+                else tx
+            )
+            for tx in value["raw_transactions"]
+        ]
+        return super(Transaction, cls).from_dict(new_value)
+
+    def to_dict(self: Self) -> Dict[str, Any]:
+        """
+        Returns the dictionary representation of a Batch.
+
+        Returns:
+            The dictionary representation of a Batch.
+        """
+        tx_dict = super().to_dict()
+        tx_dict["raw_transactions"] = [
```

Comment:
Help me understand this question: Why are you manually unpacking/packing the `RawTransactions` field? There are other "inner-object" (in rippled parlance)/ `NestedModel` fields which do not have this special code in their `from_dict` and `to_dict` method definitions. (`AuthAccount`, `BatchSigner` are two examples).

Aren't inner-objects handled by the JSON library automatically? Do we need to spell out arbitrary depth nesting of the JSON object structure?

(The below structure is my mental model of the `RawTransactions` STArray inner object)

```
RawTransactions: [
    {
      RawTransaction: {
        TransactionType: "OfferCreate",
        ...
      }
    },
    {
      RawTransaction: {
        TransactionType: "Payment",
        ...
      }
    },
  ]
```

Reply:
You could have implemented this idea by making use of `NestedModel`.
```
class RawTransaction(NestedModel):
    raw_transaction: Transaction = REQUIRED # type: ignore

class Batch(Transaction):
    """Represents a Batch transaction."""

    raw_transactions: List[RawTransaction] = REQUIRED  # type: ignore
    # instead of the current design
    # raw_transactions: List[Transaction] = REQUIRED  # type: ignore
    
    # rest of the code
```

I don't see how this lessens the user-experience. As a bonus, you wouldn't have had to unpack/pack the raw-transactions yourself.